### PR TITLE
Get server environment from application config and add poison to application list

### PR DESCRIPTION
**Pull request by muhmi**
#1 Get server environment name from application config or from Mix.env - works with exrm

#2 Add poison to application list so exrm can find it

### CODE REVIEW

File: lib/honeybadger/utils.ex [hunk 1]
```diff
@@ -14,4 +14,17 @@ defmodule Honeybadger.Utils do
...
     |> Enum.join(".")
   end
+
+  @doc """
+  Resolve environment name from Application configuration or from Mix.env
+  """
+  def environment_name do
+    case Application.get_env(:honeybadger, :mix_env, nil) do
+      nil ->
+        Mix.env
```
Comment (joshuap) [inline lib/honeybadger/utils.ex]: Per #8 I'm pretty sure we shouldn't be relying on `Mix` at runtime. @rbishop can you confirm when you have the chance?

Comment (rbishop) [inline lib/honeybadger/utils.ex]: Yes we've had a problem with that in the past. That was the original cause of #8. If you build a release Mix gets compiled out.

Comment (joshuap) [inline lib/honeybadger/utils.ex]: OK thanks, that's what I thought.

**Comment by joshuap** [comment]
@muhmi thanks for submitting; we can't merge the fix for the mix environment unfortunately because we can't depend on `Mix.env` (I know the lib currently does this, but it's something I need to change). I am working on a fix for that, and will also add the environment_name setting at that time. Sorry about that!

Regarding adding `:poison` to the application list, I think @rbishop was still looking into that. If you want you could submit a separate PR for that separately (although from what I read, we're still unsure if it's necessary).

**Comment by muhmi** [comment]
ok, I will use my fork until that. It works with exrm with the changes.
Mix.env wont be called if the application config declares environment name.
Its fine to reference the module as long as you dont try to call it ;) but
I suppose it would be better not to depent on it at all.

Thanks for the quick reply!

**Comment by rbishop** [comment]
Thanks! I'm currently reading through this to try and get the low down on library Applications: 

http://erlang.org/pipermail/erlang-questions/2010-April/050506.html

**Comment by rbishop** [comment]
According to the documentation for Application.start/2, we should add `:poison` to the `included_applications` since it is an Application that does not have processes, or rather do not need to be started.

http://elixir-lang.org/docs/v1.0/elixir/Application.html#start/2

**Comment by muhmi** [comment]
Sorry to comment on a closed request but I came across this issue with exrm:
https://github.com/bitwalker/exrm/issues/147

After I tried to move things in to `:included_applications`.

Found this bit in the comments:

"You should use applications unless you explicitly do not want the application in question to be started, but there is almost never a reason for this unless you plan to manage the start manually. Even "library" applications (I make a distinction here, but really there isn't one) can be started, though the operation is effectively a no-op. You do not want to put things in included_applications just because they do not have an application which needs to be started"

This from https://github.com/bitwalker who is the guy behind exrm. 

So I guess putting `:poison` to `:applications` is the way to go.

**Comment by joshuap** [comment]
@muhmi thanks! Is that a change we still need to make in this lib? If so, could you create a new issue for it?

**Comment by jonatandahora** [comment]
I'm also having problems with Mix.env always reporting as :prod as described in #8 and @muhmi fork works for me.
There isn't any way this could be merged or that you guys could add a fix that doesn't rely on Mix?

**Pull request closed without merging**